packages/master-detail-layout/src/vaadin-master-detail-layout.js

```diff
@@ -352,6 +352,17 @@ class MasterDetailLayout extends SlotStylesMixin(ResizeMixin(ElementMixin(Themab
     `;
   }
 
+  /** @protected */
+  async connectedCallback() {
```

Comment:
This approach might not to be enough to make it work with the Flow component. I was testing with this Flow component branch that adds the `setDetails` call: https://github.com/vaadin/flow-components/tree/refactor/mdl-set-details-method. When opening http://localhost:8080/vaadin-master-detail-layout/demo/category/Fruit/product/7 in the demo app it uses a transition initially. Maybe we have to add some custom logic to the Flow component?